dot/build_spec_test.go

```diff
@@ -0,0 +1,301 @@
+				genesis: &genesis.Genesis{
+					Name:            "test",
+					ID:              "ID",
+					ChainType:       "chainType",
+					ProtocolID:      "protocol",
+					ConsensusEngine: "babe",
+				},
+			},
+			want: `
+{
+    "name": "test",
+    "id": "ID",
+    "chainType": "chainType",
+    "bootNodes": null,
+    "telemetryEndpoints": null,
+    "protocolId": "protocol",
+    "genesis": {},
+    "properties": null,
+    "forkBlocks": null,
+    "badBlocks": null,
+    "consensusEngine": "",
+    "codeSubstitutes": null
+}`,
+		},
+		{
+			name: "normal conditions",
+			buildSpec: &BuildSpec{
+				genesis: &genesis.Genesis{
+					Name:               "test",
+					ID:                 "ID",
+					ChainType:          "chainType",
+					Bootnodes:          []string{"node1", "node2"},
+					TelemetryEndpoints: []interface{}{"endpoint"},
+					ProtocolID:         "protocol",
+					Genesis:            genesis.Fields{},
+					Properties:         map[string]interface{}{"key": "value"},
+					ForkBlocks:         []string{"1", "2"},
+					BadBlocks:          []string{"3", "4"},
+					ConsensusEngine:    "babe",
+					CodeSubstitutes:    map[string]string{"key": "value"},
+				},
+			},
+			want: `
+{
+    "name": "test",
+    "id": "ID",
+    "chainType": "chainType",
+    "bootNodes": [
+        "node1",
+        "node2"
+    ],
+    "telemetryEndpoints": null,
+    "protocolId": "protocol",
+    "genesis": {},
+    "properties": {
+        "key": "value"
+    },
+    "forkBlocks": null,
+    "badBlocks": null,
+    "consensusEngine": "",
+    "codeSubstitutes": null
+}`,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := tt.buildSpec.ToJSON()
+			assert.ErrorIs(t, err, tt.err)
+			assert.Equal(t, strings.TrimSpace(tt.want), string(got))
+		})
+	}
+}
+
+func TestBuildFromDB(t *testing.T) {
+	// initialise node (initialise state database and load genesis data)
+	cfg := NewTestConfig(t)
+	cfg.Init.Genesis = utils.GetGssmrGenesisRawPathTest(t)
+	builder := nodeBuilder{}
+	err := builder.initNode(cfg)
+	require.NoError(t, err)
+
+	tests := []struct {
+		name string
+		path string
+		want *BuildSpec
+		err  error
+	}{
+		{name: "normal conditions", path: cfg.Global.BasePath,
+			want: &BuildSpec{genesis: &genesis.Genesis{
+				Name:       "Gossamer",
+				ID:         "gssmr",
+				Bootnodes:  []string{},
+				ProtocolID: "/gossamer/gssmr/0",
+				Genesis: genesis.Fields{
+					Raw:     map[string]map[string]string{},
+					Runtime: map[string]map[string]interface{}{},
+				},
+			}}},
+		{name: "invalid db path", path: t.TempDir(),
+			err: errors.New("cannot start state service: failed to create block state: cannot get block 0: Key not found")},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := BuildFromDB(tt.path)
+
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+			if tt.want != nil {
```

Comment:
`tt.want != nil` this check isn't really required. we don't have a test case where want is nil. Even when we have a case where `tt.want` is nil, I think we could proceed without this check.

Same goes for other places where this check exists in this file.

Reply:
The test "invalid db path" has want is nil, so I need this check here because since the result (got) is nil it throws a SIGABRT when trying to call got.genesis....